### DISCUSSION TITLE
讨论关于相机相对于光机的转角问题

### DISCUSSION THREAD

**Discussion by HaiMianBBao** [Open3DV/Xema#6] [Q&A] [answered]
嗨，你们好，很高兴发现这个开源项目，也很感谢你们团队做出这么棒的产品并把它开源出来，我是一个刚刚接触3d结构光的小白，我想借此项目来搭建一个3d结构光相机。我在参考你们3d相机的结构时发现到3d相机的光机是朝着正前方向的，然后相机是朝着光机那边转了一点角度的，我想知道这个转的角度有哪些讲究嘛？

**Comment by vic-w** [comment, marked as answer]
这是个很有探讨价值的问题。各家3D相机厂商设置的光机和相机的角度各有不同。有的是光机朝着正前方，有的是相机朝正前方，有的是光机和相机都向内倾斜。不过总体来讲，倾斜的原因就是为了使相机和光机的视野范围重叠的更多一些。比如XEMA-S这个型号，它的工作距离在400-1000mm之间，我们让光机和相机的光轴相交在800mm处，这样它在其工作距离的范围内重叠的部分会多一些。

除了使视野重叠更多这个目的之外，两个镜头角度的设置对于3D测量的精度也是有影响的。

不论是光机还是相机，镜头成像最清晰的位置总是在焦平面上。其中一个倾斜会导致两个镜头的焦平面是不重合的。如果我们用3D相机测量一个类平面的物体（比如一块电路板），那么这个被测物体要么放在光机的焦平面上，要么放在相机的焦平面上。总之没办法同时让光机和相机同时成像清晰。当然，有些厂商是用沙姆镜头（移轴镜头）让两个镜头同时清晰，但那意味着要付出额外的成本。

如果我们不打算用沙姆镜头，那么我们就必须在相机和光机中选择一个，放弃对绝对清晰的追求。到底放弃光机还是放弃相机呢？我目前的观点是，保证相机的清晰是最重要的，光机稍微模糊一点也无所谓。也就是说相机应该是朝着正前方的，而光机应该是倾斜的。XEMA项目后面会推出的P系列（高精度）3D相机，两个镜头的角度就是这么设置的。

原因有两方面：
1. 对于光机来说，其投影的条纹是正弦条纹。假设镜头失焦，那么相当于对这个正弦条纹施加了一个高斯模糊。而正弦函数和一个高斯函数叠加，结果仍然是正弦函数。而且正弦的相位也不发生变化。只是正弦的幅度稍微减小了一些。对测量的精度造成不影响。
2. 对于相机来说，如果失焦问题就严重的多了。相机的失焦相当于对采集到的图片施加了一个高斯模糊。如果被测物体本身的颜色有很大反差（比如一张印有黑白图案的纸），那么在黑白交界的地方，白色那边的亮度就会因为失焦的原因蔓延到黑色那一边。这就打破了“相移测量法”的根本假设：”每个像素的深度值只取决于其像素本身的亮度“。从成像的点云来看，在黑白交界的地方，一边会偏近，另一边偏远。

所以当追求精度绝对准确的时候，我们应当尽量保证相机这一边的镜头是清晰的。

但如果我们的被测物体实际上不是一个类平面的物体，而是一个具有立体形状的物体，那么其表面总会有一些区域处在焦平面之外。这时似乎我们无论怎样都无法保证成像绝对清晰，那么讨论如何设计镜头的角度似乎就意义不大了。

以上是我个人的对于精度的一些理解，不一定对。希望朋友们能提出宝贵的意见，帮助XEMA项目越做越好。
